contracts/Core/RewardManager.sol

```diff
@@ -100,7 +100,7 @@ contract RewardManager is Initializable, Constants, RewardManagerParams, IReward
         Structs.Staker memory thisStaker = stakeManager.getStaker(stakerId);
         uint32 epochLastRevealed = voteManager.getEpochLastRevealed(stakerId);
         if (epochLastRevealed != 0 && epochLastRevealed < epoch - 1) {
-            stakeManager.setStakerAge(epoch, thisStaker.id, 0);
+            stakeManager.setStakerAge(epoch, thisStaker.id, 0, AgeChanged.MissedEpoch);
```

Comment:
Hey I think there is issue with this age mech itself 
How there can be AgeChanged.InactivityPenalty and AgeChanged.MissedEpoch both ?

so at first in givePenalties, we call giveInactivityPenalties which says penalise this much for age 
= no of epcochs missed * 10000 

then we come to this if 
which directly reduce it to 0 

isn't this double ?
what was purpose of this if ?

Reply:
@razor-network/engineering

Reply:
looks like this if is not needed, what was its purpose 
anyways we are thinking removing this step where we reduce age directly to 0 
so should we remove this if entirely ? @hrishikeshio @SkandaBhat 

Related Issue https://github.com/razor-network/contracts/issues/587

Reply:
yes but the if and return is still needed. else the following calculation will not be correct.